components/OAuth2.0/OauthAPICall.c

```diff
@@ -0,0 +1,148 @@
+#include "OauthAPICall.h"
+#include "authorization.h"
+
+static const char *TAG = "HTTP";
+APIBuffer_t *ServiceBuffer;
+char Base64Credintials[SMALL_BUF] = {0};
+
+void APICallInit(APIBuffer_t *APIBuffer)
+{
+    ServiceBuffer = APIBuffer;
+    sprintf(Base64Credintials, "Basic %s", BASE64_CREDINTIALS);
+}
+
+static esp_err_t HttpEventHandler(esp_http_client_event_t *evt) 
+{
+    static int totalLen = 0;
+    // Create the queue and semaphore
+
+    switch (evt->event_id) 
+    {
+        case HTTP_EVENT_ERROR:
+            break;
+        case HTTP_EVENT_ON_CONNECTED:
+            break;
+        case HTTP_EVENT_HEADER_SENT:
+            break;
+        case HTTP_EVENT_ON_HEADER:
+            // Handle HTTP header received, if needed
+            break;
+        case HTTP_EVENT_ON_DATA:
+            if (totalLen + evt->data_len < SUPER_BUF)                                       // check if receved data is not larger than buffer size 
+            {   
+                memcpy(ServiceBuffer->MessageBuffer + totalLen, evt->data, evt->data_len);  // copy received data to the end of previous received data
+                totalLen += evt->data_len;                                                  // update pointer to the end of copied data
+            }
+            break;
+        case HTTP_EVENT_ON_FINISH:
+            ServiceBuffer->MessageBuffer[totalLen] = '\0';
+                                            // write 0 to the end of string
+            break;
+        case HTTP_EVENT_DISCONNECTED:
+            xSemaphoreGive(ServiceBuffer->ResponseReadyFlag);                        // give semaphore to notify that data is ready  
+            totalLen = 0;                                                                   // reset contect length counter
+            break;
+        case HTTP_EVENT_REDIRECT:
+            ESP_LOGI(TAG, "redirected");
+            break;
+    }
+    return ESP_OK;
+}
+
+/**
+ * @brief This function sends a request to the Service login API to exchange an authorization code for an access token.
+ * @param[in] Code is parameter that we give it before .
+ * @param[in] SizeCode The size of the authorization code.
+ * @return This function does not return a value.
+ */
+void SendTokenRequest(char *Code)
+{  
+    esp_http_client_config_t clientConfig = {
+        .url = "https://accounts.spotify.com/api/token",                            
+        .host = "accounts.spotify.com",
+        .path = "/api/token",
+        .method = HTTP_METHOD_POST,
```

Comment:
Your main duty was making a struct that can give this parameter from user! but at now in all where you use HARD CODE!